src/cockpit/389-console/src/monitor.jsx

```diff
@@ -580,6 +589,25 @@ export class Monitor extends React.Component {
                 });
     }
 
+    getDBEngine () {
+        const cmd = [
+            "dsconf", "-j", "ldapi://%2fvar%2frun%2fslapd-" + this.props.serverId + ".socket",
+            "backend", "config", "get"
+        ];
+        log_cmd("getDBEngine", "Get DB Implementation", cmd);
+        cockpit
+                .spawn(cmd, { superuser: true, err: "message" })
+                .done(content => {
+                    const config = JSON.parse(content);
+                    const attrs = config.attrs;
+                    if ('nsslapd-backend-implement' in attrs) {
+                        this.setState({
+                            dbEngine: attrs['nsslapd-backend-implement'][0],
+                        });
+                    }
+                });
+    }
+
```

Comment:
If this function fails, I think, we need to report an error in a `.fail(error` section.